dotnet/samples/Concepts/Filtering/PIIDetectionWithFilters.cs

```diff
@@ -0,0 +1,404 @@
+            TestConfiguration.AzureOpenAI.Endpoint,
+            TestConfiguration.AzureOpenAI.ApiKey);
+
+        // Add logging
+        var logger = this.LoggerFactory.CreateLogger<PIIDetectionWithFilters>();
+        builder.Services.AddSingleton<ILogger>(logger);
+
+        // Add Microsoft Presidio Text Analyzer service and configure HTTP client for it
+        builder.Services.AddHttpClient<PresidioTextAnalyzerService>(client => { client.BaseAddress = new Uri("http://localhost:5001"); });
+
+        // Add prompt filter to analyze rendered prompt for PII before sending it to LLM.
+        // It's possible to change confidence score threshold value from 0 to 1 during testing to see how the logic will behave.
+        builder.Services.AddSingleton<IPromptRenderFilter>(sp => new PromptAnalyzerFilter(
+            sp.GetRequiredService<ILogger>(),
+            sp.GetRequiredService<PresidioTextAnalyzerService>(),
+            scoreThreshold: 0.5));
+
+        var kernel = builder.Build();
+
+        // Example 1: Use prompt with PII
+        var result1 = await kernel.InvokePromptAsync("John Smith drivers license is AC432223");
+        logger.LogInformation("Result: {Result}", result1.ToString());
+
+        /* 
+        Prompt: John Smith drivers license is AC432223
+        Entity type: PERSON. Score: 0.85
+        Entity type: US_DRIVER_LICENSE. Score: 0.6499999999999999
+        Result: Prompt contains PII information. Operation is canceled. 
+        */
+
+        // Example 2: Use prompt without PII
+        var result2 = await kernel.InvokePromptAsync("Hi, can you help me?");
+        logger.LogInformation("Result: {Result}", result2.ToString());
+
+        /*
+        Prompt: Hi, can you help me?
+        Result: Of course! I'm here to help. What do you need assistance with?
+        */
+    }
+
+    /// <summary>
+    /// Use Presidio Text Anonymizer to detect PII information in prompt and update the prompt by following specified rules before sending it to LLM.
+    /// Text Anonymizer API: https://microsoft.github.io/presidio/api-docs/api-docs.html#tag/Anonymizer.
+    /// </summary>
+    [Fact]
+    public async Task PromptAnonymizerAsync()
+    {
+        var builder = Kernel.CreateBuilder();
+
+        // Add Azure OpenAI chat completion service
+        builder.AddAzureOpenAIChatCompletion(
+            TestConfiguration.AzureOpenAI.ChatDeploymentName,
+            TestConfiguration.AzureOpenAI.Endpoint,
+            TestConfiguration.AzureOpenAI.ApiKey);
+
+        // Add logging
+        var logger = this.LoggerFactory.CreateLogger<PIIDetectionWithFilters>();
+        builder.Services.AddSingleton<ILogger>(logger);
+
+        // Add Microsoft Presidio Text Analyzer service and configure HTTP client for it. Text Analyzer results are required for Text Anonymizer input.
+        builder.Services.AddHttpClient<PresidioTextAnalyzerService>(client => { client.BaseAddress = new Uri("http://localhost:5001"); });
```

Comment:
You've already registered this above.

Reply:
Correct, but this is different test with new kernel builder. I was thinking to put it in common place and re-use, but this example file is already big enough with Presidio HTTP service clients, so I decided to keep it simple and configure it again. If there will be a 3rd test with the same logic, I will definitely put it in common place to respect Rule of Three :)